### ISSUE TITLE
VPN Issue - HMAC Authentication Error

### ISSUE THREAD

**Issue by jrac86** [dperson/openvpn-client#59]
I seem to be having an issue where requests are timing out. I thought it was the DNS issue that you mentioned in the readme, but even after setting the DNS to google's servers, I still have connectivity issues. I looked at the vpn logs and saw this error several times:

Authenticate/Decrypt packet error: packet HMAC authentication failed

Could this be causing my issue? Any ideas on how to fix this?

Thank you in advance!

**Comment by jrac86** [comment]
Mon Apr 10 19:20:59 2017 UDP link local: (not bound)
Mon Apr 10 19:20:59 2017 UDP link remote: [AF_INET]216.151.180.35:1194
Mon Apr 10 19:21:02 2017 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1542', remote='link-mtu 1570'
Mon Apr 10 19:21:02 2017 WARNING: 'cipher' is used inconsistently, local='cipher BF-CBC', remote='cipher AES-256-CBC'
Mon Apr 10 19:21:02 2017 WARNING: 'auth' is used inconsistently, local='auth SHA1', remote='auth SHA256'
Mon Apr 10 19:21:02 2017 WARNING: 'keysize' is used inconsistently, local='keysize 128', remote='keysize 256'
Mon Apr 10 19:21:02 2017 [vpn] Peer Connection Initiated with [AF_INET]216.151.180.35:1194
Mon Apr 10 19:21:04 2017 Option 'explicit-exit-notify' in [PUSH-OPTIONS]:5 is ignored by previous <connection> blocks
Mon Apr 10 19:21:04 2017 WARNING: INSECURE cipher with block size less than 128 bit (64 bit).  This allows attacks like
Mon Apr 10 19:21:04 2017 WARNING: INSECURE cipher with block size less than 128 bit (64 bit).  This allows attacks like
Mon Apr 10 19:21:04 2017 WARNING: cipher with small block size in use, reducing reneg-bytes to 64MB to mitigate SWEET32
Mon Apr 10 19:21:04 2017 TUN/TAP device tun0 opened
Mon Apr 10 19:21:04 2017 do_ifconfig, tt->did_ifconfig_ipv6_setup=0
Mon Apr 10 19:21:04 2017 /sbin/ip link set dev tun0 up mtu 1500
Mon Apr 10 19:21:04 2017 /sbin/ip addr add dev tun0 172.21.90.209/23 broadcast 172.21.91.255
Mon Apr 10 19:21:04 2017 Initialization Sequence Completed
***Mon Apr 10 19:21:25 2017 Authenticate/Decrypt packet error: packet HMAC authentication failed
Mon Apr 10 19:21:45 2017 Authenticate/Decrypt packet error: packet HMAC authentication failed***
Mon Apr 10 19:21:46 2017 [vpn] Inactivity timeout (--ping-restart), restarting
Mon Apr 10 19:21:46 2017 /sbin/ip addr del dev tun0 172.21.90.209/23
Mon Apr 10 19:21:46 2017 SIGUSR1[soft,ping-restart] received, process restarting

**Comment by jrac86** [comment]
So I tried this again with a different VPN provider and it worked. The working provider's ovpn file included a ca, cert, and key whereas the one that didnt work only had a ca in it's .ovpn. I looked everywhere to see if I was missing a config file or something, but it seems like this provider (newshosting) only gives out the ca. Could someone maybe explain what the difference is between the ca and cert? Newshoting gives a .ovpn file with ca in it and a .crt file, but when I compare them, they look the exact same.
